packages/builder/src/components/automation/SetupPanel/RowSelector.svelte

```diff
@@ -86,130 +193,175 @@
     return value
   }
 
-  const onChange = (e, field, type) => {
-    let newValue = {
-      ...value,
-      [field]: coerce(e.detail, type),
-    }
-    dispatch("change", newValue)
+  const isFullWidth = type => {
+    return (
+      attachmentTypes.includes(type) ||
+      type === FieldType.JSON ||
+      type === FieldType.LONGFORM
+    )
   }
 
-  const onChangeSetting = (field, key, value) => {
-    let newField = {}
-    newField[field] = {
-      [key]: value,
-    }
-
-    let updatedFields = {
-      ...meta?.fields,
-      ...newField,
+  const onChange = update => {
+    const customizer = (objValue, srcValue) => {
```

Comment:
Is this basically doing a recursive deep merge? Hard to tell from the code but I think that's what is going on...